src/game/engine/cargo.cpp

```diff
@@ -18,66 +18,63 @@
 #include "foot.h"
 #include "target.h"
 
+/**
+ * @brief
+ *
+ * @address 0x004623D0
+ */
 void CargoClass::Attach(FootClass *object)
 {
-    // TODO Needs FootClass
-#ifdef GAME_DLL
-    void (*func)(CargoClass *, FootClass *) = reinterpret_cast<void (*)(CargoClass *, FootClass *)>(0x004623D0);
-    func(this, object);
-#else
-    /*if (object != nullptr) {
-        object->Limbo();
-
-        // finding the last object in the list?
-        ObjectClass *next_obj = nullptr;
-        for (next_obj = object->Next; next_obj->Next; next_obj = next_obj->Next);
+    if (object != nullptr) {
 
-        if (next_obj != nullptr) { // temp
-            next_obj->Next = Object;
+        object->Limbo();
+        ObjectClass *obj = nullptr;
+        for (obj = object->Get_Next(); obj != nullptr && obj->Get_Next(); obj = obj->Get_Next()) {
```

Comment:
Write this as `for (obj = object->Get_Next(); obj != nullptr && obj->Get_Next(); obj = obj->Get_Next());` for a single line statement that iterates the object list, its tidier.